src/WebJobs.Script/OutOfProc/Http/HttpFunctionInvocationDispatcher.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Eventing;
+using Microsoft.Azure.WebJobs.Script.Rpc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using FunctionMetadata = Microsoft.Azure.WebJobs.Script.Description.FunctionMetadata;
+
+namespace Microsoft.Azure.WebJobs.Script.OutOfProc
+{
+    internal class HttpFunctionInvocationDispatcher : IFunctionDispatcher
```

Comment:
It seems like the process management logic is being duplicated (ex: methods around starting/stopping, HttpWorkerErrorEvent and HttpWorkerRestartEvent), but I'm not sure if it needs to be.

It seems like we should be able to separate out the process management portions from the invocation / setup. Maybe this a change for a future iteration, but I think that it's going to be difficult to maintain two separate code paths for process management, when what needs to be done is largely the same.

Reply:
I agree. yes this is one part i did not refactor yet. Opened https://github.com/Azure/azure-functions-host/issues/4977